attribution/experiment_logger.py

```diff
@@ -196,39 +311,64 @@ def print_sentence_attribution(self, score_agg: Literal["mean", "sum", "last"] =
                 self.df_experiments["exp_id"] == exp_id, "unit_definition"
             ].values[0]
 
-            sentence_data = {
+            total_data = {
                 "exp_id": exp_id,
                 "attribution_strategy": attr_strat,
                 "perturbation_strategy": perturbation_strategy,
                 "unit_definition": unit_definition,
             }
-            sentence_data.update(
+            total_data.update(
                 {f"token_{i+1}": token_attr for i, token_attr in enumerate(token_attrs)}
             )
+            totals.append(total_data)
 
-            sentences.append(sentence_data)
-
-        df_sentences = pd.DataFrame(sentences)
-        self.pretty_print(df_sentences)
+        df_totals = pd.DataFrame(totals)
+        self.pretty_print(df_totals)
 
     def print_attribution_matrix(
         self,
-        exp_id: int,
+        exp_id: int = -1,
```

Comment:
It's not "no value" - it's the experiment index. So -1 gives us the last (as in most recent) experiment, just like indexing into a list.

Reply:
from the code:

if exp_id == -1:
            exp_id = self.df_experiments["exp_id"].max()

Reply:
Oh yea I realise that, but it feels kinda "fake" to me. 

I was trying to say that, as a developer seeing `-1` as the value I would assume it's being passed used for indexing, and that -2 would be also be valid. But it isn't because `-1` is actually a magic number that triggers an if condition.

However I hadn't seen the first time that `None` is also a meaningful value later in the code so we can't use that.

How about we support actual negative indexing with something like:
```python
if exp_id < 0:
    exp_id = self.df_experiments["exp_id"].max() + 1 + exp_id
```